pkg/operator/operator.go

```diff
@@ -194,13 +194,25 @@ func NewOperator() (context.Context, *Operator) {
 		return []string{o.(*v1beta1.NodeClaim).Status.ProviderID}
 	}), "failed to setup nodeclaim provider id indexer")
 	lo.Must0(mgr.GetFieldIndexer().IndexField(ctx, &v1beta1.NodeClaim{}, "spec.nodeClassRef.apiVersion", func(o client.Object) []string {
-		return []string{o.(*v1beta1.NodeClaim).Spec.NodeClassRef.APIVersion}
+		nc := o.(*v1beta1.NodeClaim)
```

Comment:
Can we validate what the result of this happening is? I'm assuming that we don't crash, but do we clean things up and throw errors properly?

Reply:
Can we add a test into Kwok or AWS provider (or both) for validating that we don't crash here?

Reply:
In this case, there's no error to throw since we're failing open. This is consistent with other places in the project that we check the `NodeClassRef`. I can add a test in the AWS provider since we don't have e2e testing wired fully up with KWOK yet.